### DISCUSSION TITLE
Not able to select the relational fields with type-graphql

### DISCUSSION THREAD

**Discussion by hardikmodi1** [MichalLytek/typegraphql-prisma#325] [Q&A]
I have defined following model in schema.prisma file 

```
model User {
  id            String   @id @default(uuid())
  createdAt     DateTime @default(now())
  updatedAt     DateTime @updatedAt
  name          String?
  email         String?
  entityId      String?
  entity        Entity?  @relation(fields: [entityId], references: [id])
}

model Entity {
  id    String     @unique @default(uuid())
  city  String
  type  EntityType
  users User[]

  @@unique([city, type])
}
```

And ran prisma gen command, now I want to write my own resolver instead of using builtin resolvers provided by type-grapql.

So I have following resolver

```
import { User } from '@generated/type-graphql/models/User'

@Resolver()
export class MeResolver {
  @Query(() => User, { nullable: true })
  async me(@Ctx() ctx: Context): Promise<User | null> {
    const user = await ctx.prisma.user.findFirst({
      where: { id: ctx.req.user?.id },
      include: {
        entity: true,
      },
    })
    if (user) {
      console.log(user)
      return user
    }
    throw new ForbiddenError()
  }
}
```

I can see `entity` present when I print user in console. but in graphql query I am not able to select `entity` field. Is there anything I am doing wrong here?

Thanks for any help or suggestions!

**Comment by MichalLytek** [comment]
You need to import and use relation resolvers or write your own ones.

**Reply by hardikmodi1** [reply]
Hey @MichalLytek,
Thanks for the reply

Currently I am solving this by redefining the user type and adding entity to User type. This is working fine. So this works or I have to add relation resolver?
